lib/rate/bloc/rate_bloc.dart

```diff
@@ -32,22 +36,51 @@ class RateBloc extends Bloc<RateEvent, RateState> {
     RateEvent event,
   ) async* {
     if (event is RateTalk) {
-      try {
-        final canRateTime = event.talk.endTime.subtract(Duration(minutes: 5));
-        if (DateTime.now().isAfter(canRateTime)) {
-          _ratingsRepository.rateTalk(event.talk.id, event.rating.toInt());
-          _rating = event.rating;
-          yield TalkRatedState();
-        } else {
-          yield RatingTalkToEarlyErrorState();
-        }
-      } catch (e, s) {
-        logger.errorException(e, s);
-        yield RatingTalkErrorState();
-      }
+      yield* handleRateTalkEvent(event);
+    } else if (event is ReviewTalk) {
+      yield* handleReviewTalkEvent(event);
     } else if (event is FetchRateTalk) {
-      _rating = _ratingsRepository.myRatingOfTalk(event.talk.id)?.toDouble();
-      yield TalkRateFetched();
+      yield* handleFetchRateTalk(event);
+    }
+  }
+
+  Stream<RateState> handleRateTalkEvent(RateTalk event) async* {
+    try {
+      if (canRateTalk(event.talk)) {
+        _ratingsRepository.rateTalk(event.talk.id, event.rating.toInt());
+        _rating = event.rating;
+        yield TalkRatedState();
+      } else {
+        yield RatingTalkToEarlyErrorState();
+      }
+    } catch (e, s) {
+      logger.errorException(e, s);
+      yield RatingTalkErrorState();
+    }
+  }
+
+  Stream<RateState> handleFetchRateTalk(FetchRateTalk event) async* {
+    _rating = _ratingsRepository.myRatingOfTalk(event.talk.id)?.toDouble();
+    _review = _ratingsRepository.myReviewOfTalk(event.talk.id);
+
+    yield TalkRateFetched();
+  }
+
+  Stream<RateState> handleReviewTalkEvent(ReviewTalk event) async* {
+    try {
+      if (canRateTalk(event.talk)) {
+        _ratingsRepository.reviewTalk(event.talk.id, event.review);
+        _review = event.review;
+        yield TalkRatedState();
+      } else {
+        yield RatingTalkToEarlyErrorState();
+      }
+    } catch (e, s) {
+      logger.errorException(e, s);
+      yield RatingTalkErrorState();
     }
   }
+
+  bool canRateTalk(Talk talk) =>
+      DateTime.now().isAfter(talk.endTime.subtract(Duration(minutes: 5)));
```

Comment:
Can we make it configurable from remote config maybe? Maybe in another PR...

Reply:
Sure. I'll make another PR to fix that.